### ISSUE TITLE
Incorrect usage of "optional" in argument type

### ISSUE THREAD

**Issue by willemvdb42** [NilsJPWerner/autoDocstring#228]
When writing the following function

```
def my_function(arg: int = 3) -> int:
    return arg
```

autoDocstring generates the following docstring automatically (using Google style docstrings):
```
def my_function(arg: int = 3) -> int:
    """
    [summary]

    Args:
        arg (int, optional): [description]. Defaults to 3.

    Returns:
        int: [description]
    """
    return arg
```

The type for `arg` is incorrect. It is not optional. Optional does not mean there is a default value. Optional means the argument can also be None. So if the type of `arg` is `(int, optional)`, then the corresponding type hint would have to be `Optional[int]` or `Union[int, None]`, which is not the case here.

See https://docs.python.org/3/library/typing.html#special-forms for more info. (Look for Optional right below.)

Ps: I'm using v0.4.1 on vscode. (Which vscode is somehow assuring me is the latest version.)

**Comment by jpy-git** [comment]
Yes I always have this issue and would be great to see it fixed! 😄

As @willemvdb42 mentions the `(<type>, optional)` in the [google docstring style examples](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) is because the param is either None or the other type, not that it has a default (although often None is used as a default arg).

I use the [darglint](https://github.com/terrencepreilly/darglint) tool to detect discrepancies between the type hints and the docstring and I always have to fix this error manually:
<img width="855" alt="image" src="https://user-images.githubusercontent.com/80432516/159133114-1bfa9480-8018-4a6c-a63d-b3435571eb37.png">

**Comment by jepperaskdk** [comment]
Perhaps this is a different issue, but `Optional[str] = None` also becomes `(Optional[str], optional)` - I guess it is redundant

**Comment by dvorst** [comment]
> Perhaps this is a different issue, but `Optional[str] = None` also becomes `(Optional[str], optional)` - I guess it is redundant

I am having the same issue and I'm responding to the thread to ensure it stays open.

**Comment by Lambda-Flipper** [comment]
Still same issue here with latest autodocstring + darglint2.

**Comment by bsridatta** [comment]
No updates here?

**Comment by jepperaskdk** [comment]
To summarize, the current use of `optional` is incorrect. It appears to be used for anything that is a `kwarg` (has a default value).

The only scenarios where we want to see `type, optional`, is if the type is possibly `None`.

So either the `Optional` type:
```
def foo(x: Optional[int]):
    pass
```

Or any union that involves `None`:
```
def foo(x: Union[int, None]):
    pass

def foo(x: Union[int, str, None]):
    pass

# PEP 604 version
def foo(x: int, y: int | str | None):
    pass
```

Is this assumption correct?

In all cases, is the type unwrapped to prevent redundency? So for the above scenarios:
`Optional[int]`: `(Optional[int], optional)` or `(int, optional)`?
`Union[int, None]`: `(Union[int, None], optional)` or `(int, optional)`?
`Union[int, str, None]`: `(Union[int, str, None], optional)` or `(Union[int, str], optional)`?
`int | str | None`: `(int | str | None, optional)` or `(int | str, optional)`?
